tests/simulation/testParticleGenerator.cpp

```diff
@@ -99,4 +101,48 @@ TEST_F(ParticleGeneratorTest, CorrectCuboid)
     EXPECT_EQ(
         container.getContainer().size(),
         numParticlesWidth * numParticlesHeight * numParticlesDepth);
+}
+
+// Test if the corners of the cuboid are correct
+TEST_F(ParticleGeneratorTest, CorrectCorners)
+{
+    std::array<double, 3> corner1 = origin;
+    std::array<double, 3> corner2 = { origin[0] + (numParticlesWidth - 1) * spacing,
+                                      origin[1],
+                                      origin[2] };
+    std::array<double, 3> corner3 = { origin[0],
+                                      origin[1] + (numParticlesHeight - 1) * spacing,
+                                      origin[2] };
+    std::array<double, 3> corner4 = { origin[0] + (numParticlesWidth - 1) * spacing,
+                                      origin[1] + (numParticlesHeight - 1) * spacing,
+                                      origin[2] };
+    std::array<double, 3> corner5 = { origin[0],
+                                      origin[1],
+                                      origin[2] + (numParticlesDepth - 1) * spacing };
+    std::array<double, 3> corner6 = { origin[0] + (numParticlesWidth - 1) * spacing,
+                                      origin[1],
+                                      origin[2] + (numParticlesDepth - 1) * spacing };
+    std::array<double, 3> corner7 = { origin[0],
+                                      origin[1] + (numParticlesHeight - 1) * spacing,
+                                      origin[2] + (numParticlesDepth - 1) * spacing };
+    std::array<double, 3> corner8 = { origin[0] + (numParticlesWidth - 1) * spacing,
+                                      origin[1] + (numParticlesHeight - 1) * spacing,
+                                      origin[2] + (numParticlesDepth - 1) * spacing };
+
+    std::array<std::array<double, 3>, 8> corners = { corner1, corner2, corner3, corner4,
+                                                     corner5, corner6, corner7, corner8 };
+    std::array<bool, 8> found = { false, false, false, false, false, false, false, false };
+
+    for (auto& p : container) {
+        std::array<double, 3> pos = p.getX();
+        for (int i = 0; i < 8; i++) {
+            if (ArrayUtils::L2Norm(pos - corners[i]) < 1e-6) {
+                found[i] = true;
+            }
+        }
+    }
+
+    for (int i = 0; i < 8; i++) {
+        EXPECT_TRUE(found[i]);
+    }
```

Comment:
Here you could just check in the for-loop like so:
```c++
for (int i=0; i < 8; i++) EXPECT_LT(ArrayUtils::L2Norm(pos - corners[i]) < 1e-6)
```
and get rid of the second for-loop and the `found` array :)

Reply:
You cannot really do this, as this would test each particle being close to all corners.